useraccount/internal/ports/driverframework.go

```diff
@@ -1,25 +1,14 @@
 package ports
 
-/** we should probably define our data schema in go
-// I believe the protoc compiler will do that for us but not sure
-// I will just use placeholder here
-*/
- */
+import (
+	"context"
 
-type Person struct {
+	"github.com/cb-technologies/existe-id/useraccount/useraccount/internal/adapters/framework/driver/grpc/pb"
+)
 
-}
-
-type PersonId struct {
-
-}
-
- type FindPersonKey struct {
-
-}
-
-type APIPort interface {
-	AddNewPersonInfo(person Person) error
-	EditPersonInfo(pid PersonId) error
-	FindPersonInfo(key FindPersonKey) (Person, error)
+type GRPCPort interface {
+	Run()
+	AddNewPersonInfo(ctx context.Context, req *pb.PersonInfo) (*pb.Response, error)
+	UpdatePersonInfo(ctx context.Context, req *pb.EditPersonInfoParameters) (*pb.Response, error)
+	FindPersonInfo(ctx context.Context, req *pb.UUID) (*pb.PersonInfo, error)
```

Comment:
I was just wondering if we should modify the interface of FindPersonInfo to return a response as well. Make it so that all our API call will always return a Response pb.Response type. 
something like FindPersonInfo(ctx context.Context, req *pb.UUID) (*pb.Response,*pb.PersonInfo, error)

Reply:
We can but it's going to be a little bit redundant... 

We have pb.Response for the two previous call because naturally they don't return anything and we want to make sure that the operation was successful; that's the purpose of having pb.Response

So for FindPersonInfo, we already know if the operation is successful by the presence of the date (pb.PersonInfo). So checking again for a response status is redundant.